src/main/java/net/glowstone/block/itemtype/ItemWrittenBook.java

```diff
@@ -30,6 +31,10 @@ public void rightClickBlock(
     }
 
     private void openBook(GlowPlayer player) {
-        player.getSession().send(new PluginMessage("MC|BOpen", EMPTY));
+        ByteBuf buf = Unpooled.buffer();
+        GlowBufUtils.writeHand(buf, player.getMainHand());
```

Comment:
This VarInt isn't for left or right, but for main or offhand. You can use a book in your offhand if there is nothing in your main hand when you right click.